### ISSUE TITLE
Project ERROR: Unknown module(s) in QT: webkitwidgets

### ISSUE THREAD

**Issue by qiuzhu** [goldendict/goldendict#1086]
I install QT 5.11.2 on ubuntu 16.04 , and when qmake the project, it throws error:
`Project ERROR: Unknown module(s) in QT: webkitwidgets`

how to solve the problem

**Comment by Abs62** [comment]
GD require QtWebkit module. Find it or build from source.

**Comment by qiuzhu** [comment]
QtWebkit module is removed from QT 5.11，how to find this Qtwebkit ？I Google it,but failed

**Comment by Abs62** [comment]
IDK. QtWebkit source you can find [here](https://download.qt.io/official_releases/qt/5.9/5.9.1/submodules/).

**Comment by SunYongshuai** [comment]
I also have this problem. And when i build QtWebKit alone, it shows
"Project ERROR: WebKit requires SQLite. Either make it available via pkg-config, set $SQLITE3SRCDIR or build WebKit under qt5.git."
so, is there a simple way to build this project with recent Qt?

**Comment by vedgy** [comment]
Some distributions package the latest version of Qt and QtWebkit. For instance [the Arch Linux package](https://www.archlinux.org/packages/extra/x86_64/qt5-webkit/) uses the source code package from [here](https://github.com/annulen/webkit/releases). So for Arch Linux users building Goldendict with the latest Qt and QtWebkit versions is trivial. If anyone manages to build Goldendict with a non-system Qt and QtWebkit versions, they are welcome to post the steps in the [Goldendict wiki](https://github.com/goldendict/goldendict/wiki).
What is so important for Goldendict in recent Qt versions that you would suffer the overhead and possible inconsistencies of running two versions of Qt in one system?

**Comment by a17r** [comment]
Are there any plans to port to QtWebEngine? Qt5WebKit is deprecated for a long time, not part of official Qt releases since 5.6 and meanwhile dropped wherever possible.

**Comment by qiuzhu** [comment]
in the wiki, there is a compiled portable edition, but it is too old, I hope the latest release will be maintained. because install by myself is painstaking.

**Comment by vedgy** [comment]
@qiuzhu, why not build goldendict with your Ubuntu 16.04 system Qt version 5.5.1? This build and installation process is documented in Goldendict's README and should be straightforward.

**Comment by qiuzhu** [comment]
@vedgy 
thank you very much, because I install the latest QT 5.11.2, and that is a tragedy. how can I make it compiled successfully without changing the QT edition?

**Comment by vedgy** [comment]
@qiuzhu, the process is described at length here: https://github.com/goldendict/goldendict/blob/master/README.md. You don't have to install any libraries from outside the official Ubuntu repositories. Start with the *Installing External Deps on Ubuntu Linux for Qt5* section.
**Alternatively** start with *Installing External Deps on Ubuntu Linux* if you prefer the older Qt4. You only have to run a single *apt-get* command - installing either Qt5 or Qt4 dependencies.

**Comment by SunYongshuai** [comment]
I have just compiled it with qt5.12.0 (lastest) at ubuntu 18.04.1.
You can add qtwebkit submodule to yout QT to compile it. Something in my memory:
1) Download qtwebkit soure code to compile it with&for your current qt at first. It may be error,but Just get .so files for using.  http://code.qt.io/cgit/qt/qtwebkit.git/
2) Add include files etc. from old(lastest as you find) compiled by others, 
3) Compile goldendict as usual.

Because of no detail recorded, just this info for you. And my problems still there, i give up the new version i compiled.

**Comment by qiuzhu** [comment]
@vedgy 
I try the method following step by step, and the same error.

well, this is my default setting QT as is shown in the below snapshot:
![11](https://user-images.githubusercontent.com/13781719/50481548-302b8a80-0a1d-11e9-9ea1-4049981ea38e.png)

**Comment by vedgy** [comment]
@SunYongshuai, [Qt WebKit](https://packages.ubuntu.com/xenial/libqt5webkit5-dev) is available in official Ubuntu repositories. No need to compile it manually when using system Qt version.
@qiuzhu, your qmake uses the new Qt version you have installed in your home directory. Either switch back to the system one by invoking something like /usr/bin/qmake (not sure if it will work so simply), or better uninstall Qt 5.11 if you don't need it for any other application.

**Comment by SunYongshuai** [comment]
![image](https://user-images.githubusercontent.com/25148301/50482188-64547a80-0a20-11e9-9108-16a561a09d74.png)
Qt-make in Ubuntu18.04.1 is just 5.9.5.

**Comment by vedgy** [comment]
Yes, it's 5.9 in 18.04 and 5.5 in 16.04. These versions are perfectly suitable to build Goldendict with. I've actually compiled and run a recent Goldendict version with Qt 5.1/5.2 in Ubuntu 14.04. Why do you need the latest Qt version?

**Comment by SunYongshuai** [comment]
@vedgy I just like to use latest Qt version. Maybe it is better to suit for recently applications.
